smartdashboard/views.py

```diff
@@ -342,3 +343,203 @@ def __init__(
+        self.memory_table_element.dataframe(
+            dframe.tail(1), use_container_width=True, hide_index=True
+        )
+
+
+class ClientView(ViewBase):
+    """View class for client section of the Database Telemetry page"""
+
+    def __init__(self, shard: t.Optional[Shard]) -> None:
+        """Initialize a ClientView
+
+        :param shard: Selected shard
+        :type shard: t.Optional[Shard]
+        """
+        self.shard = shard
+        self.client_table_element = DeltaGenerator()
+        self.client_graph_element = DeltaGenerator()
+
+    def update_shard(self, new_shard: t.Optional[Shard]) -> None:
+        """Update selected shard
+
+        This is called after a new shard is selected
+        in the dashboard to keep displayed data in sync.
+
+        :param new_shard: Selected shard
+        :type new_shard: Optional[Shard]
+        """
+        if new_shard is not None:
+            self.shard = new_shard
+
+    def update(self) -> None:
+        """Update client table and graph elements for selected shard"""
+        if self.shard is not None:
+            try:
+                client_df = pd.read_csv(self.shard.client_file)
+                counts_df = pd.read_csv(self.shard.client_count_file)
+                self.update_client_table(client_df)
+                self.update_client_graph(counts_df)
+            except FileNotFoundError:
+                self.client_table_element.info(
+                    f"Client data was not found for {self.shard.name}"
+                )
+
+    def update_client_table(self, dframe: pd.DataFrame) -> None:
+        """Update client table for selected shard
+
+        :param dframe: DataFrame with client data
+        :type dframe: pandas.DataFrame
+        """
+        self.client_table_element.dataframe(
+            dframe, use_container_width=True, hide_index=True
+        )
+
+    def update_client_graph(self, dframe: pd.DataFrame) -> None:
+        """Update client graph for selected shard
+
+        :param dframe: DataFrame with client data
+        :type dframe: pandas.DataFrame
+        """
+        chart = (
+            alt.Chart(dframe)
+            .mark_line()
+            .encode(x="Time", y="Client Count", tooltip=["Time", "Client Count"])
+            .properties(
+                height=500,
+                title=alt.TitleParams("Total Client Count", anchor="middle"),
+            )
+        )
+
+        self.client_graph_element.altair_chart(chart, use_container_width=True)
+
+
+class OrchestratorSummaryView(ViewBase):
+    """View class for orchestrator summary section of the Database Telemetry page"""
+
+    def __init__(self, orchestrator: t.Optional[Orchestrator]) -> None:
+        """Initialize an OrchestratorSummaryView
+
+        :param orchestrator: Selected orchestrator
+        :type orchestrator: t.Optional[Orchestrator]
+        """
+        self.orchestrator = orchestrator
+        self.status_element = DeltaGenerator()
```

Comment:
This is trickier than I expected. Because `OrchestratorView` has updating logs, random log elements are written if I try to use it in place of `OrchestratorSummaryView`. I'm going to chat with you offline about this one!

Reply:
Per offline conversion, we decided this was not trivially feasible, and if we do decided to address it should be its own ticket. Resolving!